.github/workflows/build-and-push.yaml

```diff
@@ -50,10 +54,28 @@ jobs:
       run: |
         make build-image
 
+    - name: Write signing key to disk (only needed for `cosign sign --key`)
+      run: echo "${{ secrets.COSIGN_KEY }}" > cosign.key
+
     - name: Push images
       run: |
         make push-image
 
+    - name: Sign images
```

Comment:
As discussed, add `if: ${{ startsWith(github.ref, 'refs/tags/') }}` bellow `-name` to sign only tags